Library/ENV/4.3/cc

```diff
@@ -194,6 +196,27 @@ class Cmd
   end
 
   def keep?(path)
+    # The logic in this method will eventually become the default,
+    # but is currently opt-in.
+    return keep_orig?(path) unless ENV["HOMEBREW_EXPERIMENTAL_FILTER_FLAGS_ON_DEPS"]
+
+    # first two paths: reject references to Cellar or opt paths
+    # for unspecified dependencies
+    if path.start_with?(cellar)
+      %r[#{cellar}/([\w\-_\+]+)] =~ path
+      $1 && @deps.include?($1)
+    elsif path.start_with?(opt)
+      %r[#{opt}/([\w\-_\+]+)] =~ path
+      $1 && @deps.include?($1)
+    elsif path.start_with?(prefix)
+      true
+    else
+      !path.start_with?("/opt", "/sw", "/usr/X11")
```

Comment:
Yeh. I think making the change in CI and then fixing these issues as they come up might be a good approach. That's like what we've done with the sandbox CC @xu-cheng